core/src/main/java/org/apache/calcite/rel/externalize/RelJson.java

```diff
@@ -557,6 +569,14 @@ private Object toJson(RexNode node) {
         map.put("type", toJson(call.getType()));
         final List<@Nullable Object> list = jsonBuilder().list();
         for (RexNode operand : call.getOperands()) {
+          if (!CALCITE_5614_FIXED) {
```

Comment:
Move this check into the case statement after CORREL_VARIABLE

```
    case SEARCH:
      if (!CALCITE_5614_FIXED) {
        return toJson(RexUtil.expandSearch(new RexBuilder(new JavaTypeFactoryImpl()), null, node));
      }
```
should be all we need